### ISSUE TITLE
Does not show any file or folder after loading file manager.

### ISSUE THREAD

**Issue by tanbir2013** [simogeo/Filemanager#382]
I am installed the file manager but it does not show any folder or files in my root folder.It shows only this picture:

![img 351](https://cloud.githubusercontent.com/assets/9824429/9426672/a8a485a8-4972-11e5-9a9e-2f0a2ab46732.jpg)

my server root is www/_publicHtml

```javascript
{
		"_comment": "IMPORTANT : go to the wiki page to know about options configuration https://github.com/simogeo/Filemanager/wiki/Filemanager-configuration-file",
    "options": {
        "culture": "en",
        "lang": "php",
        "theme": "flat-oil",
        "defaultViewMode": "grid",
        "autoload": true,
        "showFullPath": true,
        "showTitleAttr": false,
        "browseOnly": false,
        "showConfirmation": true,
        "showThumbs": true,
        "generateThumbnails": true,
        "searchBox": true,
        "listFiles": true,
        "fileSorting": "default",
        "chars_only_latin": true,
        "dateFormat": "d M Y H:i",
        "serverRoot": true,
        "fileRoot": "www/_publicHtml/",
        "baseUrl": "false",
        "logger": false,
		"logfile": "D:/server/www/_publicHtml/test/FileMan/filemanager_log.txt",
		
        "capabilities": ["select", "download", "rename", "delete", "replace"],
        "plugins": []
    },
    "security": {
        "allowFolderDownload": false,
        "allowChangeExtensions": false,
        "allowNoExtension": false,
        "uploadPolicy": "DISALLOW_ALL",
        "uploadRestrictions": [
            "jpg",
            "jpe",
            "jpeg",
            "gif",
            "png",
            "svg",
            "txt",
            "pdf",
            "odp",
            "ods",
            "odt",
            "rtf",
            "doc",
            "docx",
            "xls",
            "xlsx",
            "ppt",
            "pptx",
            "csv",
            "ogv",
            "mp4",
            "webm",
            "m4v",
            "ogg",
            "mp3",
            "wav",
            "zip",
            "rar"
        ]
    },
    "upload": {
        "multiple": true,
        "number": 5,
        "overwrite": false,
        "imagesOnly": false,
        "fileSizeLimit": 16
    },
    "exclude": {
        "unallowed_files": [
            ".htaccess",
            "web.config"
        ],
        "unallowed_dirs": [
            "_thumbs",
            ".CDN_ACCESS_LOGS",
            "cloudservers"
        ],
        "unallowed_files_REGEXP": "/^\\./",
        "unallowed_dirs_REGEXP": "/^\\./"
    },
    "images": {
        "imagesExt": [
            "jpg",
            "jpe",
            "jpeg",
            "gif",
            "png",
            "svg"
        ],
        "resize": {
        	"enabled":true,
        	"maxWidth": 1280,
            "maxHeight": 1024
        }
    },
    "videos": {
        "showVideoPlayer": true,
        "videosExt": [
            "ogv",
            "mp4",
            "webm",
            "m4v"
        ],
        "videosPlayerWidth": 400,
        "videosPlayerHeight": 222
    },
    "audios": {
        "showAudioPlayer": true,
        "audiosExt": [
            "ogg",
            "mp3",
            "wav"
        ]
    },
    "pdfs": {
        "showPdfReader": true,
        "pdfsExt": [
            "pdf",
            "odp"
        ],
	    "pdfsReaderWidth": "640",
        "pdfsReaderHeight": "480"	
    },
    "edit": {
        "enabled": true,
        "lineNumbers": true,
        "lineWrapping": true,
        "codeHighlight": false,
        "theme": "elegant",
        "editExt": [
            "txt",
            "csv"
        ]
    },
    "customScrollbar": {
    	"enabled": true,
    	"theme": "inset-2-dark",
    	"button": true
    },
    "extras": {
        "extra_js": [],
        "extra_js_async": true
    },
    "icons": {
        "path": "images/fileicons/",
        "directory": "_Open.png",
        "default": "default.png"
    },
    "url": "https://github.com/simogeo/Filemanager",
    "version": "2.0.0-dev"
}
```

**Comment by simogeo** [comment]
And what about : 

```javascript
        "serverRoot": false,
        "fileRoot": "D:/server/www/_publicHtml/",
```

By the way, if it still does not work, you can enable logfile and verify path settings to see what's happening. Also, have a look to apache log files or open Development Console

**Comment by Jimmi08** [comment]
Exactly the same problem here. Fresh install, if this helps:
![image](https://cloud.githubusercontent.com/assets/5429548/9449375/795fdcee-4aa2-11e5-95b8-0cbe52f07116.png)
But paths are correct, files are there.

**Comment by simogeo** [comment]
What's happening if you directly open jquery file into the browser ?

**Comment by Jimmi08** [comment]
I see its code, so I know that path is correct.

**Comment by simogeo** [comment]
!!!??? 

1. I just downloaded from https://github.com/simogeo/Filemanager/archive/master.zip

2. copied *filemanager.config.js.default* to *filemanager.config.js*

3. opened in a browser ... and **it just works** !

**Comment by Jimmi08** [comment]
No, exactly the same picture as in first comment above.I use latest Chrome. When I click on element inspector, and console, I see errors I have already sent. Oh, I tried it with CKE editor.

**Comment by simogeo** [comment]
Can you try to call the FM directly on browser ? Can you try with FF or another browser ?

**Comment by tanbir2013** [comment]
i already fix up.but I can not attach with my laravel 5.0 framework?
Do you have idea about it?

On 8/25/15, Simon Georget <notifications@github.com> wrote:
> Can you try to call the FM directly on browser ? Can you try with FF or
> another browser ?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/simogeo/Filemanager/issues/382#issuecomment-134491043


-- 
 Best Regards,
 Engr.Tanbir Hasan
 Lecturer,Dhaka Engineering Institute
 Dhaka,Bangladesh

**Comment by simogeo** [comment]
No idea, see the foloowing how-to, it could help you : https://github.com/simogeo/Filemanager/wiki/Integrate-the-Filemanager-with-Laravel

**Comment by himpatel** [comment]
How about 
"serverRoot": true,
"fileRoot": "",

At least this works in Asp.net (config: lang = "ashx") mode. I had the same issue and after making this change, the UI showed up files from "userFiles" folder.

**Comment by hbekkouche** [comment]
I have the same problem, What about connectors I dont find it ?

**Comment by anhnnp** [comment]
Hi, I have the same error, I using xampp on window 7 and I access http://localhost/.../filemanager/index.html still ok. But I clone to project centos, it does not show any folder or files in my root folder. It show same that your picture tanbir2013. 
I enable logfile but not error. Who can help me?

**Comment by simogeo** [comment]
What's the developer tool > network panel is saying ?

Any warning ?

**Comment by anhnnp** [comment]
I have carefully read your documents without finding where the problem lies. I've set the folder permissions 777

![screenshot from 2016-10-08 16-22-45](https://cloud.githubusercontent.com/assets/10054554/19212173/6a030b64-8d76-11e6-9977-6b890d4bf67a.png)
![screenshot from 2016-10-08 16-41-22](https://cloud.githubusercontent.com/assets/10054554/19212174/6bc171b6-8d76-11e6-9925-c438bd15b28f.png)
_**File log server apache php**_
`[Sat Oct 08 16:22:35.249202 2016] [:error] [pid 5469] [client ::1:45676] PHP Warning:  date(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected the timezone 'UTC' for now, but please set date.timezone to select your timezone. in /var/www/html/maygiatcongnghiep247.com/assets/filemanager/connectors/php/filemanager.class.php on line 1538, referer: http://localhost/maygiatcongnghiep247.com/assets/filemanager/
[Sat Oct 08 16:22:35.249211 2016] [:error] [pid 5469] [client ::1:45676] PHP Stack trace:, referer: http://localhost/maygiatcongnghiep247.com/assets/filemanager/
[Sat Oct 08 16:22:35.249216 2016] [:error] [pid 5469] [client ::1:45676] PHP   1. {main}() /var/www/html/maygiatcongnghiep247.com/assets/filemanager/connectors/php/filemanager.php:0, referer: http://localhost/maygiatcongnghiep247.com/assets/filemanager/
[Sat Oct 08 16:22:35.249221 2016] [:error] [pid 5469] [client ::1:45676] PHP   2. include_once() /var/www/html/maygiatcongnghiep247.com/assets/filemanager/connectors/php/filemanager.php:55, referer: http://localhost/maygiatcongnghiep247.com/assets/filemanager/
[Sat Oct 08 16:22:35.249226 2016] [:error] [pid 5469] [client ::1:45676] PHP   3. Filemanager->__construct() /var/www/html/maygiatcongnghiep247.com/assets/filemanager/connectors/php/user.config.php:54, referer: http://localhost/maygiatcongnghiep247.com/assets/filemanager/
[Sat Oct 08 16:22:35.249231 2016] [:error] [pid 5469] [client ::1:45676] PHP   4. Filemanager->__log() /var/www/html/maygiatcongnghiep247.com/assets/filemanager/connectors/php/filemanager.class.php:113, referer: http://localhost/maygiatcongnghiep247.com/assets/filemanager/
[Sat Oct 08 16:22:35.249235 2016] [:error] [pid 5469] [client ::1:45676] PHP   5. date() /var/www/html/maygiatcongnghiep247.com/assets/filemanager/connectors/php/filemanager.class.php:1538, referer: http://localhost/maygiatcongnghiep247.com/assets/filemanager/
`
**_File log filemanager_**
```
[08/10/2016 09:22:35]#::1#Filemanager::enableLog - Log enabled (in /var/www/html/maygiatcongnghiep247.com/filemanager_log.txt file)
[08/10/2016 09:22:35]#::1#Filemanager::__construct $this->root value /var/www/html/maygiatcongnghiep247.com/assets/filemanager/
[08/10/2016 09:22:35]#::1#Filemanager::__construct $this->path_to_files /var/www/html/maygiatcongnghiep247.com/public/images/
[08/10/2016 09:22:35]#::1#Filemanager::__construct $this->doc_root value /var/www/html
[08/10/2016 09:22:35]#::1#Filemanager::__construct $this->separator value images
[08/10/2016 09:22:35]#::1#substr path : /var/www/html/maygiatcongnghiep247.com/public/images/
[08/10/2016 09:22:35]#::1#real path : /var/www/html/maygiatcongnghiep247.com/public/images/
```
**_filemanager.config.json_**
`options": {
        "culture": "en",
        "lang": "php",
        "theme": "flat-turquoise",
        "defaultViewMode": "grid",
        "autoload": true,
        "showFullPath": false,
        "showTitleAttr": false,
        "browseOnly": false,
        "showConfirmation": true,
        "showThumbs": true,
        "generateThumbnails": true,
        "cacheThumbnails": true,
        "searchBox": true,
        "listFiles": true,
        "fileSorting": "default",
        "quickSelect": false,
        "chars_only_latin": true,
        "splitterWidth": 200,
        "splitterMinWidth": 200,
        "dateFormat": "d M Y H:i",
        "serverRoot": true,
        "fileRoot": "maygiatcongnghiep247.com/public/images/",
        "baseUrl": false,
        "logger": true,
        "logfile": "/var/www/html/maygiatcongnghiep247.com/filemanager_log.txt",
        "capabilities": ["select", "download", "rename", "delete", "replace"],
        "plugins": []
    }`
Thank so much!

**Comment by simogeo** [comment]
Form developer tool > networ panel, have a look on the response of the following call (or similar) : 

> http://yourserverurl.com/connectors/php/filemanager.php?path=%2Fuserfiles%2F&mode=getfolder&showThumbs=true&time=929

And see if you don't have unwanted content such as php warning ....

**Comment by anhnnp** [comment]
I look tab network panel request 

> http://localhost/maygiatcongnghiep247.com/assets/filemanager/connectors/php/filemanager.php?path=%2Fmaygiatcongnghiep247.com%2Fpublic%2Fimages%2F&config=filemanager.config.json&mode=getfolder&showThumbs=true&time=450

This is my picture 
![screenshot from 2016-10-08 21-18-56](https://cloud.githubusercontent.com/assets/10054554/19213704/2a9db178-8d9d-11e6-98b6-321b4a5d73cf.png)
I really do not know where the error, although on my win7 are running normally.
I downloaded your FileManager put in public_html, visit the url: (http://localhost/FileManager/index.html) are running normally. I still do not know why, if you find a bug please just help me. Thanks a lot!

**Comment by simogeo** [comment]
click on Response tab (see below) and copy / paste the result : 

![2a9db178-8d9d-11e6-98b6-321b4a5d73cf](https://cloud.githubusercontent.com/assets/154323/19214583/84af885a-8d87-11e6-837b-209a4b44251f.png)

**Comment by anhnnp** [comment]
Okey, thank you so much!
![screenshot from 2016-10-09 15-17-10](https://cloud.githubusercontent.com/assets/10054554/19219065/9a62dca4-8e33-11e6-88d4-c0c59654e2e6.png)

> A PHP Error was encountered
Severity: Warning
Message: date(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected the timezone 'UTC' for now, but please set date.timezone to select your timezone.
Filename: php/filemanager.class.php
Line Number: 298

The log file apache server has notify this message error but I think it doesn't affect "using date() do not safe and only select date.zonetime my zonetime". I have add `date_default_timezone_set('UTC');` into construct() method in `filemanager.class.php`. And then your filemanager have worked.

Although Windows still runs well, do not know if anyone has encountered this bug on linux yet? Just add specific date.zonetime utc.

**Comment by simogeo** [comment]
Thanks for letting me know ... I propose a fix here : fe511d23d200e739aaf1e4d20263f20cf1b2d524

now integrated in the main repo.

**Comment by tanbir2013** [comment]
thanks to you.

On 10/9/16, Simon Georget <notifications@github.com> wrote:
> Thanks for letting me know ... I propose a fix here :
> fe511d23d200e739aaf1e4d20263f20cf1b2d524
>
> now integrated in the main repo.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/simogeo/Filemanager/issues/382#issuecomment-252478521


-- 
 Best Regards,
 Engr.Tanbir Hasan
 Lecturer,Dhaka Engineering Institute
 Dhaka,Bangladesh
